### PR TITLE
bigz/marketBidAskTwapCrank.ts

### PR DESCRIPTION
**Pull request by 0xbigz**
`ts-node -T src/index.ts --mark-twap-crank --run-once`

### CODE REVIEW

File: src/bots/marketBidAskTwapCrank.ts [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+		return combinedList;
+	}
+
+	private async tryTwapCrank() {
+		await this.init();
+		await this.initDlob2();
+
+		const ixs = [];
+
+		const crankMarkets = [0, 1, 2, 5, 8];
```
Comment (crispheaney) [inline src/bots/marketBidAskTwapCrank.ts:174]: intentionally not running it for all mkts?

File: src/bots/marketBidAskTwapCrank.ts [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+	public async healthCheck(): Promise<boolean> {
+		let healthy = false;
+		await this.watchdogTimerMutex.runExclusive(async () => {
+			healthy =
+				this.watchdogTimerLastPatTime > Date.now() - 2 * this.defaultIntervalMs;
+		});
+		return healthy;
+	}
+
+	private async initDlob2() {
```
Comment (crispheaney) [inline src/bots/marketBidAskTwapCrank.ts:132]: `initDlob`?

File: src/bots/marketBidAskTwapCrank.ts [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+		this.driftEnv = driftEnv;
+	}
+
+	public async init() {
+		logger.info(`${this.name} initing`);
+		this.lookupTableAccount =
+			await this.driftClient.fetchMarketLookupTableAccount();
+
+		// initialize userMap instance
+		this.userMap = new UserMap(
```
Comment (crispheaney) [inline src/bots/marketBidAskTwapCrank.ts:94]: are the user maps every shared btwn bots?

File: src/bots/marketBidAskTwapCrank.ts (hunk 1 above)
Comment (0xbigz) [reply src/bots/marketBidAskTwapCrank.ts:174]: yeah should make a couple bundles, was just testing

Comment (wphan) [reply src/bots/marketBidAskTwapCrank.ts:174]: Possible to make this crank all markets by default? Maybe user can pass an arg to for subset

File: src/bots/marketBidAskTwapCrank.ts (hunk 3 above)
Comment (0xbigz) [reply src/bots/marketBidAskTwapCrank.ts:94]: not sure..

File: src/bots/marketBidAskTwapCrank.ts (hunk 1 above)
Comment (0xbigz) [reply src/bots/marketBidAskTwapCrank.ts:174]: just made it loop over every perp market in chunks

**Pull request merged**